gflownet/utils/buffer.py

```diff
@@ -259,11 +259,47 @@ def make_data_set(self, config):
         )
         return df, {"x": samples, "energy": energies}
 
-    def compute_stats(self, data):
+    @staticmethod
+    def compute_stats(data):
         mean_data = data["energies"].mean()
         std_data = data["energies"].std()
         min_data = data["energies"].min()
         max_data = data["energies"].max()
         data_zscores = (data["energies"] - mean_data) / std_data
         max_norm_data = data_zscores.max()
         return mean_data, std_data, min_data, max_data, max_norm_data
+
+    @staticmethod
+    def select(data_dict: dict, n: int, mode: str = "permutation", rng=None):
```

Comment:
If you're gonna have type hints I would be complete:

* missing `rng`
* missing return `-> list` (and ideally list of what with `List[]`

Also, I think it's time to be a bit more rigorous with docstrings. Copilot / ChatGPT will do a good job at doing a good structure / draft if you find it annoying to do :)

Reply:
By "missing rng" you mean the type is missing?

Reply:
Added docstring. Thanks for the reminder :smiling_face_with_tear: